Paltineanu Rares-Mihai/3.7 - curs4/main.c

```diff
@@ -1,48 +1,71 @@
-/*Read from the standard inputthe hexadecimal digits of aninteger hexadecimal number.
+/**Read from the standard inputthe hexadecimal digits of aninteger hexadecimal number.
 Find and display the equivalent decimal number.
 */
+///A291 - 4
+/// I = 3
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
+
+int verif(char hex[50])
+{
+    int i = strlen(hex) - 1;
+    for(int j = 0; j <= i; j++)
+    {
+        if(hex[j]>'F')
+        {
+            return 0;
+        }
+
+    }
+    return 1;
+}
+
 int main()
 {
     char hex[50];
     double dec = 0;
     int i;
     gets(hex);
-    puts(hex);
+    while(verif(hex) == 0)
+    {
+        printf("Invalid nr, please read another nr ");
+        gets(hex);
+    }
     i = strlen(hex) - 1;
 
     while(i >= 0)
     {
         if(hex[i] >= '0' && hex[i] <= '9')
-            dec = (hex[i]-'0')*pow(16,i);
+        {
+            dec = dec + (hex[i]-'0')*pow(16,strlen(hex) - i - 1);
+        }
         else
         {
             if(hex[i] == 'A')
```

Comment:
Here's an instance where a `switch` might work as well!
Or maybe, one could extract all that in a function and simply have different parameters? Much of the code seems the same